atomix/utils/src/main/java/io/atomix/utils/serializer/FallbackNamespace.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright © 2020 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.utils.serializer;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import com.google.common.collect.ImmutableList;
+import io.atomix.utils.serializer.NamespaceImpl.Builder;
+import io.atomix.utils.serializer.NamespaceImpl.RegistrationBlock;
+import java.nio.ByteBuffer;
+import org.slf4j.Logger;
+
+public class FallbackNamespace implements Namespace {
+
+  private static final Logger LOG = getLogger(FallbackNamespace.class);
+  private static final String DESERIALIZE_ERROR =
+      "Serialized bytes contained header with version but deserialization failed (will fallback to FieldSerializer):\n";
+  private static final String UNKNOWN_VERSION_ERROR =
+      "Magic byte was encountered, signalling newer version of serializer, but version {} is unrecognized. Using FieldSerializer as fallback";
+  private static final byte MAGIC_BYTE = (byte) 0xFF;
+  private static final byte VERSION_BYTE = 0x01;
+  private final NamespaceImpl fallback;
+  private final NamespaceImpl namespace;
+
+  FallbackNamespace(final NamespaceImpl fallback, final NamespaceImpl namespace) {
+    this.fallback = fallback;
```

Comment:
Name fallback is a bit misleading as it uses fallback as default for serializing. Can we give more meaningful names?

Reply:
These names were requested by the reviewer in the last review https://github.com/zeebe-io/zeebe/pull/5042#discussion_r459589513. The initial names were `legacy` and `compatible`. Can you suggest alternatives that are more meaningful?

Reply:
I think it make sense to use "fallback" in 0.25 because it is actually using it as a fallback. But here fallback is the default. `legacy` and `compatible` sounds good to me. Or just switch the places - current namespace is the new fallback.